docs/Minimal_Build__Advanced_Usage_with_NNAPI.md

```diff
@@ -1,77 +1,121 @@
- - At the 'extended' level we additionally fuse the Conv and Relu nodes using the internal ONNX Runtime FusedConv operator.
+
+- At the 'basic' level we combine the Conv and Add nodes (the addition is done via the 'B' input to Conv), we combine the MatMul and Add into a single Gemm node (the addition is done via the 'C' input to Gemm), and constant fold to remove one of the Reshape nodes. 
+- At the 'extended' level we additionally fuse the Conv and Relu nodes using the internal ONNX Runtime FusedConv operator.
 
 <img align="center" src="images/mnist_optimization.png" alt="Changes to nodes from basic and extended optimizations."/>
 
-At runtime, for each group of nodes that is assigned to it, the NNAPI EP will create an NNAPI model that replicates the processing done by the nodes in the group. It will create a function to handle executing this NNAPI model, including copying data between the CPU and NNAPI. The original nodes in the model are replaced with a single node containing this function.
+During runtime, for each group of nodes that is assigned to it, the NNAPI EP will create an NNAPI model that replicates the processing done by the nodes in the group. It will create a function to handle executing this NNAPI model, including copying data between the CPU and NNAPI. The original nodes in the model are replaced with a single node containing this function.
 
 ### Considerations when creating an NNAPI-aware ORT format model
 
-There are no optimizations available at runtime for an ORT format model, so we generally want to optimize to the highest level possible when creating the model. That usually means optimizing to the 'extended' level, as 'layout' level optimizations are not recommended (they are device specific). 
+The ONNX model intended for using with NNAPI should be optimized when creating the ORT formatted model. *Step 1* and *Step 2* optimizations should be used.
+
+*Step 2* optimizations would replace group of nodes with a function when possible. This will contribute towards improving the performance by reducing data flow between CPU and NNAPI.
+
+The 'extended' optimizations from *Step 3* may benefit the ORT model as some of the CPU nodes can be replaced by the custom ONNX operators designed for improved execution on the CPU. 
 
-If the NNAPI EP is enabled when loading the ORT format model we will run Step 2 to allow it to replace the groups of nodes it can handle with a function that executes the NNAPI model for those nodes. As the NNAPI model is created at runtime using device specific information it cannot be saved as part of the ORT format model. *NOTE:* every node in the saved ORT format model can be run using a statically registered kernel, so any node that is not taken by the NNAPI EP will be run this way.
+The 'layout' optimizations are HW specific and not applicable for execution with NNAPI.
```

Comment:
nit: it's more that layout optimizations are HW specific in general. somewhat orthogonal to whether NNAPI is used or not.

Reply:
removed "HW specific"

Reply:
I would have gone the other way :) don't use 'layout' optimizations at all as they're HW specific, whether you're using NNAPI, CoreML or the CPU EP